docs/specifications/tests/RequirementsAndNormalizationOfDomainNames.md

```diff
@@ -0,0 +1,516 @@
+characters are added to the domain name by mistake. The domain name will in this
+specification be normalized by removing such characters. In Table 3 it is
+specified what counts as white space characters. It should be pointed out that
+white space characters within the domain name are not removed, and in the end
+count as invalid characters.
+
+*Table 3: White space characters**
+
+Unicode code | Name
+:------------|:--------------------
+U+0020       | [SPACE]
+U+0009       | [CHARACTER TABULATION]
+U+00A0       | [NO-BREAK SPACE]
+U+2000       | [EN QUAD]
+U+2001       | [EM QUAD]
+U+2002       | [EN SPACE]
+U+2003       | [EM SPACE]
+U+2004       | [THREE-PER-EM SPACE]
+U+2005       | [FOUR-PER-EM SPACE]
+U+2006       | [SIX-PER-EM SPACE]
+U+2007       | [FIGURE SPACE]
+U+2008       | [PUNCTUATION SPACE]
+U+2009       | [THIN SPACE]
+U+200A       | [HAIR SPACE]
+U+205F       | [MEDIUM MATHEMATICAL SPACE]
+U+3000       | [IDEOGRAPHIC SPACE]
+U+1680       | [OGHAM SPACE MARK]
+
+#### Full stop
+
+The regular dot "." expected in domain names is a U+002E (FULL STOP), see Table 2
+above. There are other characters that may be entered instead due to the script
+setting. Table 4 lists full stop characters that are to be mapped into the
+ASCII FULL STOP ([Unicode TR 46][Unicode TR 46#Notation], section 2.3). That
+mapping must be done before any verification or checks of the dot and before
+splitting *Domain Name* into labels.
+
+*Table 4: Non-ASCII dots (Full Stops) using [Unicode] codes*
+
+Unicode code | Character | Name
+:------------|:----------|:--------------------
+U+FF0E       | ．        | [FULLWIDTH FULL STOP]
+U+3002       | 。        | [IDEOGRAPHIC FULL STOP]
+U+FF61       | ｡         | [HALFWIDTH IDEOGRAPHIC FULL STOP]
+
+#### Final dot
+
+If the domain name has one final dot it should be removed to create a consistent
+representation. The exception is the root zone which is always represented by
+the exact string ".".
+
+#### Upper case
+
+If the domain name has any letters tagged as "upper case" by the [Unicode]
+database, those should be mapped into the equivalent lower case letter. This
+applies to both ASCII (i.e. "A-Z" mapped into "a-z") in both A- and U-labels and
+non-ASCII characters found in U-labels ([RFC 5895][RFC 5895#2], section 2). This
+mapping is done before a U-label is converted to A-label. A valid U-label must
+not contain any upper case letters.
+
+For Zonemaster special rules applies to U+0049 ([LATIN CAPITAL LETTER I]) and
+U+0130 ([LATIN CAPITAL LETTER I WITH DOT ABOVE]).
+
+* [LATIN CAPITAL LETTER I] is downcased to U+0069 ([LATIN SMALL LETTER I]) also
+  in Turkish and Azeri locale, i.e. not following the special Unicode rule in
+  those locale ([Unicode SpecialCasing]).
+* Label with [LATIN CAPITAL LETTER I WITH DOT ABOVE] should be rejected since
+  normal downcasing gives a sequence not reasonable in a domain name context (see
+  "Lowercase Mapping" in [LATIN CAPITAL LETTER I WITH DOT ABOVE].
```

Comment:
There is a missing closing bracket here.

Reply:
Fixed.